### ISSUE TITLE
Migrate to CircleCI

### ISSUE THREAD

**Issue by humitos** [readthedocs/sphinx-hoverxref#106]
We are moving everything to CircleCI, so we should move this project as well.

See https://github.com/readthedocs/sphinx-notfound-page/pull/130

Be sure to keep the tests on _all the Sphinx minor versions_.

**Comment by stsewd** [comment]
> Be sure to keep the tests on all the Sphinx minor versions.

We are just supporting the latest minor version of each release for all of our extensions, except for the ones that we install by default (theme and ext). Supporting the latest minor versions makes it easy to maintain and keep up to date, plus faster CI.

**Comment by humitos** [comment]
I noted that we removed this when we migrated notfound extension (*) to CircleCI. However, I prefer to know that something is breaking in a minor Sphinx version on a PR, than from a user report. I wrote some code on these extensions that work different between minor versions (example https://github.com/humitos/sphinx-hoverxref/blob/e0df0757a2ddffe2ade29462718d9e54d8583924/hoverxref/utils.py#L10).

Due the amount of tests we have in these extensions, faster CI is not something I'm worried about.

If possible, I prefer to support all the minor Sphinx versions and if not, _know about that_ before a release.

(*) I will probably re-add them.
